modules/sdk/src/main/scala/org/tessellation/sdk/http/routes/GossipRoutes.scala

```diff
@@ -24,22 +26,23 @@ final case class GossipRoutes[F[_]: Async: KryoSerializer: Metrics](
 ) extends Http4sDsl[F] {
 
   private val prefixPath = "/rumors"
-  private val chunkSize = 10
 
   private val httpRoutes: HttpRoutes[F] = HttpRoutes.of[F] {
     case req @ POST -> Root / "peer" / "query" =>
       for {
         inquiryRequest <- req.as[PeerRumorInquiryRequest]
         inquiryOrdinals = inquiryRequest.ordinals
         localPeerIds <- rumorStorage.getPeerIds
+        rumors <- peerRumorChain(inquiryOrdinals.toList)
         additionalOrdinals = localPeerIds.diff(inquiryOrdinals.keySet).toList.map(_ -> Ordinal.MinValue)
-        result <- Ok(peerRumorStream(inquiryOrdinals.toList) ++ peerRumorStream(additionalOrdinals))
+        additionalRumors <- peerRumorChain(additionalOrdinals)
+        result <- Ok(streamFromChain(rumors ++ additionalRumors))
```

Comment:
👍🏻 

It uses O(1) `Append` under the hood. Additional savings there as well.

Reply:
True :) Chain is quite performant in many common operations.